pgaudit.c

```diff
@@ -203,6 +203,8 @@ char *auditRole = NULL;
 #define OBJECT_TYPE_FOREIGN_TABLE   "FOREIGN TABLE"
 #define OBJECT_TYPE_FUNCTION        "FUNCTION"
 
+#define OBJECT_TYPE_PARTITIONED_TABLE "PARTITIONED TABLE"
+#define OBJECT_TYPE_PARTITIONED_INDEX "PARTITIONED INDEX"
```

Comment:
I'm wondering if the users care if the table is partitioned for auditing purposes? It makes the audit log harder to parse and I'm not sure it adds value. Also, I'm worried about back-patching such a change to prior versions, especially for inherited tables, which would suddenly change types.

Thoughts?

Reply:
Hi David, I think it is necessary to audit partitioned table.
Since the partition table is used in distributed system for sharding. It is great use to analyze the 'partitioned tables' with others. Otherwise, the 'UNKNOWN' shows in the audit log which makes confusing.

Reply:
I was not suggesting we leave them as UNKNOWN. I think TABLE/INDEX would be sufficient.

More so because the output is not consistent. A partitioned table logs like:
```
select * from h_0;
NOTICE:  AUDIT: SESSION,106,1,READ,SELECT,PARTITIONED TABLE,public.h_0,select * from h_0;,<none>
```

But a partition in a table logs like:
```
select * from h_0_0;
NOTICE:  AUDIT: SESSION,107,1,READ,SELECT,TABLE,public.h_0_0,select * from h_0_0;,<none>
```
It's still part of a partitioned table, right? For auditing purposes I think knowing it is a table is sufficient. If the user wants more info they can look it up in the catalog.

Also, when dropping a partitioned table is is not classed as a partitioned table:
```
drop table h;
NOTICE:  AUDIT: SESSION,110,1,DDL,DROP TABLE,TABLE,public.h,drop table h;,<none>
NOTICE:  AUDIT: SESSION,110,1,DDL,DROP TABLE,TABLE,public.h_0,drop table h;,<none>
NOTICE:  AUDIT: SESSION,110,1,DDL,DROP TABLE,TABLE,public.h_0_0,drop table h;,<none>
NOTICE:  AUDIT: SESSION,110,1,DDL,DROP TABLE,TABLE,public.h_0_1,drop table h;,<none>
NOTICE:  AUDIT: SESSION,110,1,DDL,DROP TABLE,TABLE,public.h_1,drop table h;,<none>
```
I think consistency in classification is more important than being more accurate in some cases.

Reply:
Hi David, thanks for your detailed information. I got it.
Yes, you are correct, at least shall match with the drop action.
I will modify it.